stl/inc/ranges

```diff
@@ -7656,6 +7662,401 @@ namespace ranges {
+            // clang-format off
+            _NODISCARD_FRIEND constexpr bool operator==(const _Iterator<_IteratorConst>& _Lhs, const _Sentinel& _Rhs)
+                noexcept(noexcept(_Get_iterator_ziperator(_Lhs) == _Rhs._Inner)) // strengthened
+            {
+                // clang-format on
+                return _Get_iterator_ziperator(_Lhs) == _Rhs._Inner;
+            }
+
+            template <bool _IteratorConst>
+                requires sized_sentinel_for<_Zentinel<_IsConst>, _Ziperator<_IteratorConst>>
+            _NODISCARD_FRIEND constexpr _Iterator_sentinel_difference_type<_IteratorConst>
+                // clang-format off
+                operator-(const _Iterator<_IteratorConst>& _Lhs, const _Sentinel& _Rhs)
+                noexcept(noexcept(_Get_iterator_ziperator(_Lhs) - _Rhs._Inner)) // strengthened
+            {
+                // clang-format on
+                return _Get_iterator_ziperator(_Lhs) - _Rhs._Inner;
+            }
+
+            template <bool _IteratorConst>
+                requires sized_sentinel_for<_Zentinel<_IsConst>, _Ziperator<_IteratorConst>>
+            _NODISCARD_FRIEND constexpr _Iterator_sentinel_difference_type<_IteratorConst>
+                // clang-format off
+                operator-(const _Sentinel& _Lhs, const _Iterator<_IteratorConst>& _Rhs)
+                noexcept(noexcept(_Lhs._Inner - _Get_iterator_ziperator(_Rhs))) // strengthened
+            {
+                // clang-format on
+                return _Lhs._Inner - _Get_iterator_ziperator(_Rhs);
+            }
+        };
+
+        template <bool _IsConst>
+        static constexpr bool _Is_end_noexcept = []() noexcept {
+            if constexpr (common_range<_Maybe_const<_IsConst, _Inner_view>>) {
+                return noexcept(_Iterator<_IsConst>{_STD declval<_Maybe_const<_IsConst, zip_transform_view>&>(),
+                    _STD declval<_Maybe_const<_IsConst, _Inner_view>&>().end()});
+            } else {
+                return noexcept(_Sentinel<_IsConst>{_STD declval<_Maybe_const<_IsConst, _Inner_view>&>().end()});
+            }
+        }();
+
+        static constexpr bool _Enable_const_begin_end =
+            (range<const _Inner_view> && regular_invocable<const _Func&, range_reference_t<const _ViewTypes>...>);
+
+    public:
+        zip_transform_view() noexcept(
+            is_nothrow_default_constructible_v<_Movable_box<_Func>>&& is_nothrow_default_constructible_v<_Inner_view>) =
+            default;
+
+        constexpr explicit zip_transform_view(_Func _Function_, _ViewTypes... _Views) noexcept(
+            is_nothrow_move_constructible_v<_Movable_box<_Func>>&&
+                is_nothrow_constructible_v<_Inner_view, remove_reference_t<_ViewTypes>&&...>) // strengthened
+            : _Function(in_place, _STD move(_Function_)), _Zip(_STD move(_Views)...) {}
+
+        constexpr auto begin() noexcept(noexcept(_Iterator<false>{*this, _Zip.begin()})) // strengthened

```

Comment:
All of the overloads of `begin()`, `end()`, and `size()` should be marked `_NODISCARD`.